text/0107-construct-library-module-lifecycle.md

```diff
@@ -0,0 +1,146 @@
+
+* **Stage 1 activities:**
+
+    * **Identify key stakeholders** - Opportunistically recruit stakeholders outside the CDK team to participate in API reviews and usability studies. Potential stakeholders include: service's product manager, service's developer advocates (if any), service's software development engineer(s) (designated by service's leadership), solutions architects who specialize in service (sourced via internal mailing lists), and AWS Developer Tools developer advocates. 
+    * **Update GitHub tracking issue** - Summarize the primary use cases and required constructs as they are identified
+    * **Author RFCs [optional]** - Request for comment documents are created for large / complex designs that require explanation and robust discussion. RFCs are stored in a separate [GitHub repo](https://github.com/aws/aws-cdk-rfcs), and the module's tracking issue will include links to relevant RFCs.
+    * **Update Roadmap** - publicly indicate the module is work in progress
+        * If an RFC is written, place the module’s tracking issue on the CDK roadmap in the “researching” column to indicate we need customer feedback on the module’s design
+        * otherwise, place the module’s tracking issue on the CDK roadmap in the “Working on it“ column
+    * **Submit Pull Requests** - pull requests implementing the constructs are submitted, the API design reviewed (by module owner), and merged
+    * **Documentation** - update module’s README with explanations and code snippets that illustrate how to use new constructs as they are developed
+    * **Publish aws-xyz-experimental package** - deliver experimental constructs in their own package to clearly communicate module maturity. This package is tagged with Stability:Experimental
+    * **Observe** - monitor module utilization and GitHub issues filed against the experimental module
+        * feature requests and bugs filed against the experimental module are triaged, those that we decide to include in the module are linked in the tracking issue to indicate they are in scope
+    * **Repeat** - These activities are repeated until the module owner decides the module is feature complete and the API is ergonomic. At the end of the experimental stage, the construct library module’s GitHub tracking issue will have a description of the service, a list of targeted use cases, and a list of L2 constructs required to satisfy the use cases.
```

Comment:
When does the module owner actually decide this? Do we want a periodic check or something? Or is it the responsibility of the module owner to push his/hers modules further down the maturity pipeline?

Reply:
That's a great question, I think it's the responsibility of the module owner to know what 'done' looks like.

Reply:
I think Eli what means here is -- what triggers the area owner to take a look at the module and take a go/no-go decision?

My top of the head suggestion -- the tracking Github issue for each module should have the major use cases modeled as a checklist. Every quarter, as part of the construct lib squad prioritization, owners look at the checklists for their module and if most/all of them are checked off, they should bring it to the squad as a candidate for the next stage. Thoughts?